lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,110 @@
+          elm_test_version: 'latest'
+        }
+
+        def export
+          super
+          sh.export 'TRAVIS_ELM_VERSION', elm_version, echo: false
+          sh.export 'TRAVIS_ELM_TEST_VERSION', elm_test_version, echo: false
+        end
+
+        def configure
+          super
+
+          sh.if '! -d sysconfcpus/bin' do
+            install_sysconfcpus
+          end
+        end
+
+        def announce
+          super
+          sh.cmd 'elm --version', echo: true
+          sh.cmd 'elm-test --version', echo: true
+        end
+
+        def setup
+          super
+
+          install_elm
+          install_elm_test
+        end
+
+        def script
+          sh.cmd 'elm-format --validate .'
+          sh.cmd 'elm-test'
+        end
+
+        def setup_cache
+          if data.cache?(:elm)
+            sh.fold 'cache.elm' do
+              directory_cache.add '$HOME/.cache/elm-stuff', 'elm-stuff'
+
+              # some put their tests in ./test and others use ./tests
+              directory_cache.add '$HOME/.cache/elm-test-stuff', 'test/elm-stuff'
+              directory_cache.add '$HOME/.cache/elm-tests-stuff', 'tests/elm-stuff'
+
+              # we build sysconfcpus from source, so cache the result
+              directory_cache.add '$HOME/.cache/sysconfcpus-cache', 'sysconfcpus'
+            end
+          end
+        end
+
+        def cache_slug
+          super << '-elm-' << elm_version
+        end
+
+        private
+
+          def elm_version
+            config[:elm_version].to_s
+          end
+
+          def elm_test_version
+            config[:elm_test_version].to_s
+          end
+
+          def install_elm
+            npm_install '-g elm-use@">=1.1.0 <2.0.0"'
+            sh.cmd 'elm-use #{elm_version}', retry: true
+
+            convert_binary_to_sysconfcpus 'elm'
+            convert_binary_to_sysconfcpus 'elm-make'
+            convert_binary_to_sysconfcpus 'elm-package'
+            convert_binary_to_sysconfcpus 'elm-format'
+          end
+
+          def install_elm_test
+            npm_install '-g elm-test@#{elm_test_version}'
+
+            convert_binary_to_sysconfcpus 'elm-test'
+          end
+
+          def convert_binary_to_sysconfcpus(binary_name)
+            # Wrap the binary in a call to sysconfcpus -n 2
+            # to work around https://github.com/travis-ci/travis-ci/issues/6656
+            sh.mv "$(npm config get prefix)/bin/#{binary_name}", "$(npm config get prefix)/bin/#{binary_name}-old"
+
+            # Use printf instead of echo here.
+            # see https://github.com/rtfeldman/node-elm-compiler/pull/50
+            sh.cmd ('printf "#\041/bin/bash\n\necho \"Running ' + binary_name + ' with sysconfcpus -n 2\"\n\n$TRAVIS_BUILD_DIR/sysconfcpus/bin/sysconfcpus -n 2 ' + binary_name + '-old \"\$@\"" > $(npm config get prefix)/bin/' + binary_name)
+            sh.chmod '+x', "$(npm config get prefix)/bin/#{binary_name}"
+          end
+
+          def install_sysconfcpus
```

Comment:
These commands generate a lot of output, so I suggest putting it in a fold.

Reply:
Put behind a fold in b6697ff5bbe8de4ef3a044fe312cdb5e56a9a384